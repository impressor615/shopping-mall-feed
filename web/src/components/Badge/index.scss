@import "~@/assets/styles/variables";

.badge {
  display: inline-block;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  padding: 10px;

  color: $darkgrey;
  font-weight: bold;
  font-size: 25px;

  text-align: center;

  &-gold, &-silver, &-bronze {
    color: $white;
  }

  &-gold {
    background: $gold;
  }

  &-silver {
    background: $silver;
  }

  &-bronze {
    background: $bronze;
  }
}