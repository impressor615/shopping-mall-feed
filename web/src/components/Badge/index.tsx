import "./index.scss";

import classnames from "classnames";
import React from "react";

interface BadgeProps {
  rank: number;
}

const Badge = ({ rank }: BadgeProps) => (
  <span className={classnames(
    "badge",
    { "badge-gold": rank === 1 },
    { "badge-silver": rank === 2 },
    { "badge-bronze": rank === 3 },
  )}>
    { rank }
  </span>
);

export default Badge;
