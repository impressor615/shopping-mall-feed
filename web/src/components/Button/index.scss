@import "~@/assets/styles/variables";

.btn {
  cursor: pointer;

  padding: 10px 15px;
  border: 1px solid $secondary;

  font-size: 16px;
  color: $secondary;

  opacity: 0.6;
  transition: all 0.3s ease-out;

  &:hover {
    color: $primary;
    font-weight: bold;

    background: $white;
    opacity: 1;
  }

  &.active {
    color: $primary;
    font-weight: bold;

    background: $white;
    opacity: 1;
  }
}
