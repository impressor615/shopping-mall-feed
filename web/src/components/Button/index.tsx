import "./index.scss";

import classnames from "classnames";
import React from "react";

interface ButtonProps {
  children: React.ReactNode;
  onClick?: ()  => void;
  active?: boolean;
  [rest: string]: any;
}

const Button = ({ children, onClick, active, ...rest }: ButtonProps) => (
  <button
    className={classnames("btn", { active })}
    onClick={onClick}
    {...rest}
  >
    { children }
  </button>
);

export default Button;
