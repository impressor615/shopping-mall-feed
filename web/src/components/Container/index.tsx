import React from "react";

interface ContainerProps {
  children: React.ReactNode;
}

const Container = ({ children }: ContainerProps) => (
  <section className="container">{ children }</section>
);

export default Container;
