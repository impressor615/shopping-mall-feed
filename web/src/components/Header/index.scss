@import "~@/assets/styles/variables";

header {
  box-sizing: border-box;
  position: fixed;
  top: 0;
  width: 100%;
  max-width: 1100px;

  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px 20px;

  background: $lightgrey;
  box-shadow: 0 3px 5px $secondary;
}