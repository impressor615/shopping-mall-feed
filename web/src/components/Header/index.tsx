import "./index.scss";

import React from "react";

import HeaderFilter from "@/components/HeaderFilter";
import HeaderTitle from "@/components/HeaderTitle";

interface HeaderProps {
  filter: string;
  onClick: () => void;
}

const Header = ({ filter, onClick }: HeaderProps) => (
  <header>
    <HeaderTitle title="오늘 뭐 입지?" />
    <HeaderFilter filter={filter} onClick={onClick} />
  </header>
);

export default Header;
