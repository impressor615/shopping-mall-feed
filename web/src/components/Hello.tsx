import React from "react";
import { connect } from "react-redux";

import {
  CHANGE_COMPILER,
  CHANGE_FRAMEWORK,
} from "@/constants";

export interface HelloProps extends ConnectProps {
  compiler: string;
  framework: string;
}

class Hello extends React.PureComponent<HelloProps, {}> {
  public render() {
    const { compiler, framework } = this.props;
    return (
      <>
        <h1>Hello from {compiler} and {framework}!</h1>
        <button onClick={this.onChangeCompiler} type="button">change compiler</button>
        <button onClick={this.onChangeFramework} type="button">change framework</button>
      </>
    );
  }

  private onChangeCompiler = () => {
    const { dispatch } = this.props;
    dispatch({ type: CHANGE_COMPILER, payload: "Javascript" });
  }

  private onChangeFramework = () => {
    const { dispatch } = this.props;
    dispatch({ type: CHANGE_FRAMEWORK, payload: "CoffeeScript" });
  }
}

const mapStateToProps = ({ hello }: { hello: HelloState }) => ({
  ...hello,
});

export default connect(mapStateToProps)(Hello);
