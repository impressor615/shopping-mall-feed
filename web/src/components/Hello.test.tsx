import enzyme from "enzyme";
import React from "react";

import { Hello } from "@/components/Hello";

describe("Hello Component", () => {
  it("renders the correct text", () => {
    const wrapper = enzyme.shallow(<Hello compiler="TypeScript" framework="React" />);
    expect(wrapper.find("h1").text()).toEqual("Hello from TypeScript and React!");
  });

  it("should be matched with snapshot", () => {
    const wrapper = enzyme.shallow(<Hello compiler="TypeScript" framework="React" />);
    expect(wrapper).toMatchSnapshot();
  });
});
