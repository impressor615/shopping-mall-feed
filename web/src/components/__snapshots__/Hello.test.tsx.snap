// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Component should be matched with snapshot 1`] = `
<h1>
  Hello from 
  TypeScript
   and 
  React
  !
</h1>
`;
